pkg/sql/parser/lexer_test.go

```diff
@@ -27,6 +27,7 @@ func TestLexer(t *testing.T) {
 		{`NOT IN`, []int{NOT_LA, IN}},
 		{`NOT SIMILAR`, []int{NOT_LA, SIMILAR}},
 		{`AS OF SYSTEM TIME`, []int{AS_LA, OF, SYSTEM, TIME}},
+		{`AS OF`, []int{AS, OF}},
```

Comment:
this is merely a lexer test. 

Can you also add an exercise of `SELECT 123 AS OF FROM t` in `pkg/sql/parser/testdata/select_clauses`?

thanks